neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,275 @@
+            elif channel_info["chan_type"][channel_index][0] == "d":
+                units = "a.u."
+            # aux channel
+            else:
+                units = "V"
+
+            signal_channels.append(
+                (
+                    channel_name,
+                    channel_id,
+                    self._sampling_frequency,
+                    BINARY_DTYPE,
+                    units,
+                    1,  # no gain
+                    0,  # no offset
+                    stream_id,
+                )
+            )
+
+        signal_channels = np.array(signal_channels, dtype=_signal_channel_dtype)
+
+        stream_ids = np.unique(signal_channels["stream_id"])
+        signal_streams = np.zeros(stream_ids.size, dtype=_signal_stream_dtype)
+        signal_streams["id"] = [str(stream_id) for stream_id in stream_ids]
+        for stream_index, stream_id in enumerate(stream_ids):
+            name = stream_id_to_stream_name.get(int(stream_id), "")
+            signal_streams["name"][stream_index] = name
+
+        # No events
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes
+        spike_channels = []
+        spike_channels = np.array(spike_channels, dtype=_spike_channel_dtype)
+
+        # Put all the necessary info in the header
+        self.header = {}
+        self.header["nb_block"] = 1
+        self.header["nb_segment"] = [1]
+        self.header["signal_streams"] = signal_streams
+        self.header["signal_channels"] = signal_channels
+        self.header["spike_channels"] = spike_channels
+        self.header["event_channels"] = event_channels
+
+        # Add the minimum annotations
+        self._generate_minimal_annotations()
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+
+        # All streams have the same size so just return the raw_data size
+        return self._raw_data.size
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, stream_index, channel_indexes):
+
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self._get_signal_size(block_index, seg_index, stream_index)
+
+        if channel_indexes is None:
+            channel_indexes = slice(None)
+
+        raw_data = self._raw_data
+        return raw_data[i_start:i_stop, channel_indexes]
+
+    def _segment_t_stop(self, block_index, seg_index):
+
+        t_stop = self.metadata["status"]["t_range"][1]
+        return t_stop
+
+    def _segment_t_start(self, block_index, seg_index):
+
+        t_start = self.metadata["status"]["t_range"][0]
+        return t_start
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+
+        t_start = self.metadata["status"]["t_range"][0]
+        return t_start
+
+    #######################################
+    # Helper Functions
+
+    def read_metadata(self, fname_metadata):
```

Comment:
I like to add path (hungarian notation) to differentiate between file handles (output of f = open() and paths) but I think name works as well!

Reply:
I think the parsing is super fast, but one thing I was thinking of was we could allow think about parsing more of header in this function then in spikeinterface or in neuroconv stuff you could run this function separately if you needed separate metadata then we could add a check in the main body that if this is already run we don't need to re-run. But the thought was inchoate (because honestly this function is so small right now). Maybe next time we have a call we can brainstorm on this because I don't think I'm explaining myself super well.

As far as naming, I just didn't want to use the same name so I added the fname tag onto it. But happy to change if it helps in conceptualizing more.